htmlhint-server/src/server.ts

```diff
@@ -60,6 +65,74 @@ function getRange(error: htmlhint.Error, lines: string[]): any {
     };
 }
 
+function loadRules(HTMLHint: any, force: boolean): any {
+    if (force) {
+        rulesLoaded = false;
+    }
+
+    if (rulesLoaded || htmlhint == null) {
+        return;
+    }
+    try {
+        if (!settings.htmlhint || !settings.htmlhint.rulesDir) {
+            return;
+        }
+        let rulesDir = settings.htmlhint.rulesDir;
+        let absoluteDir = '';
+        //check absolute path
+        let exists = fs.existsSync(rulesDir);
+        if (exists) {
+            absoluteDir = rulesDir;
+        } else {
+            //relative dir, find it based on workspace roots
+            absoluteDir = path.resolve(rootFolder, rulesDir);
+            exists = fs.existsSync(absoluteDir);
+        }
+
+        if (exists) {
+            //load all rules
+            loadCustomRules(absoluteDir, HTMLHint);
+        }
+    }
+    catch (e) { }
+    finally {
+        rulesLoaded = true;
+    }
+}
+
+// load custom rles
+function loadCustomRules(rulesdir:string, HTMLHint:any):any{
+    rulesdir = rulesdir.replace(/\\/g, '/');
+    if(fs.existsSync(rulesdir)){
+        if(fs.statSync(rulesdir).isDirectory()){
+            rulesdir += /\/$/.test(rulesdir)?'':'/';
+            rulesdir += '**/*.js';
+            var arrFiles = glob.sync(rulesdir, {
+                'dot': false,
+                'nodir': true,
+                'strict': false,
+                'silent': true
+            });
+            arrFiles.forEach(function(file){
+                loadRule(file, HTMLHint);
+            });
+        }
+        else{
+            loadRule(rulesdir, HTMLHint);
+        }
+    }
+}
+
+// load rule
+function loadRule(filepath: string, HTMLHint: any): any{
+    filepath = path.resolve(filepath);
+    try{
+        var module = require(filepath);
+        module(HTMLHint);
+    }
+    catch(e){}
```

Comment:
blanket catch seems problematic here. Is this here for a reaason?

Reply:
I expect the idea is to skip a rule if it is broken. Given that this code is from htmlhint's source there might not be a way to properly report errors from there.

Can you please suggest a nice way we can report errors to the user as we have access to vscode here?